asm-to-pil/src/vm_to_constrained.rs

```diff
@@ -454,6 +456,86 @@ impl<T: FieldElement> ASMPILConverter<T> {
         res
     }
 
+    fn handle_instr_declaration_params(&mut self, m: &mut Machine<T>) {
+        for l in &mut m.links {
```

Comment:
I'm a bit lost here. Why is this function called `handle_instr_declaration_params` and then iterates over the links?
Also I think the code would be easier to understand when the iteration is pulled out of the function call, assuming the different links are not interacting (and by pulling that out of the call, you can more easily see that).

Reply:
Yeah this was not great. 
I've restructured the code handling instruction definitions (this is now part of handling external instructions), should be clearer now what's going on.